instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -0,0 +1,430 @@
+                trace.set_span_in_context(started_span)
+            )
+            Boto3SQSInstrumentor.current_span_related_to_token = started_span
+            return retval
+
+        def __iter__(self) -> Generator:
+            index = 0
+            while index < len(self):
+                yield self[index]
+                index = index + 1
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    @staticmethod
+    def _enrich_span(
+        span: Span,
+        queue_name: str,
+        conversation_id: Optional[str] = None,
+        operation: Optional[MessagingOperationValues] = None,
+        message_id: Optional[str] = None,
+    ) -> None:
+        if not span.is_recording():
+            return
+        span.set_attribute(SpanAttributes.MESSAGING_SYSTEM, "aws.sqs")
+        span.set_attribute(SpanAttributes.MESSAGING_DESTINATION, queue_name)
+        span.set_attribute(
+            SpanAttributes.MESSAGING_DESTINATION_KIND,
+            MessagingDestinationKindValues.QUEUE.value,
+        )
+        if operation:
+            span.set_attribute(
+                SpanAttributes.MESSAGING_OPERATION, operation.value
+            )
+        else:
+            span.set_attribute(SpanAttributes.MESSAGING_TEMP_DESTINATION, True)
+        if conversation_id:
+            span.set_attribute(
+                SpanAttributes.MESSAGING_CONVERSATION_ID, conversation_id
+            )
+        if message_id:
+            span.set_attribute(SpanAttributes.MESSAGING_MESSAGE_ID, message_id)
+
+    @staticmethod
+    def _safe_end_processing_span(receipt_handle: str) -> None:
+        if started_span := Boto3SQSInstrumentor.received_messages_spans.pop(
+            receipt_handle, None
+        ):
+            if (
+                Boto3SQSInstrumentor.current_span_related_to_token
+                == started_span
+            ):
+                context.detach(Boto3SQSInstrumentor.current_context_token)
+                Boto3SQSInstrumentor.current_context_token = None
+            started_span.end()
+
+    @staticmethod
+    def _extract_queue_name_from_url(queue_url: str) -> str:
+        # A Queue name cannot have the `/` char, therefore we can return the part after the last /
+        return queue_url.split("/")[-1]
+
+    def _create_processing_span(
+        self, queue_name: str, receipt_handle: str, message: Dict[str, Any]
+    ) -> None:
+        message_attributes = message.get("MessageAttributes", {})
+        links = []
+        if ctx := propagate.extract(
+            message_attributes, getter=boto3sqs_getter
+        ):
+            for item in ctx.values():
+                if hasattr(item, "get_span_context"):
+                    links.append(Link(context=item.get_span_context()))
+        span = self._tracer.start_span(
+            name=f"{queue_name} process",
+            links=links,
+            kind=SpanKind.CONSUMER,
+        )
+        with trace.use_span(span):
```

Comment:
You can combine start_span and use_span in one API: start_as_current_span() with with statement

Reply:
Nice! Done